### PR TITLE
fix: privilege escalation

### PR DESCRIPTION
**Pull request by wansir**
Signed-off-by: hongming <talonwan@yunify.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

change cluster role by self is not permitted

### CODE REVIEW

File: pkg/apiserver/iam/im.go [hunk 1]
```diff
@@ -160,6 +160,11 @@ func UpdateUser(req *restful.Request, resp *restful.Response) {
 		}
 	}
 
+	if usernameInHeader == user.Username {
+		// change cluster role by self is not permitted
```
Comment (rayzhou2017) [inline pkg/apiserver/iam/im.go]: how about admin changes itself? Still not allowed? If so, it requires a different admin to change it?

Comment (wansir) [reply pkg/apiserver/iam/im.go]: Yes, cluster role can only be modified by other admin or user manager.

**Pull request merged**